init an error array with nil

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"

var errs = []error{nil}

func main() {
    fmt.Println(errs)
}
```

Expected result:
```console
$ go run ./sample.go
[<nil>]
```

Got:
```console
$ yaegi ./sample.go
./sample.go:7:12: panic
reflect.Set: value of type interface {} is not assignable to type error
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc000142400, 0x40)
        /home/marc/sdk/go1.14/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc0000c1e10)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:320 +0xbc
panic(0xc90280, 0xc00019a6d0)
        /home/marc/sdk/go1.14/src/runtime/panic.go:967 +0x15d
github.com/containous/yaegi/interp.runCfg.func1(0xc0001315e0, 0xc000138b00)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:113 +0x20c
panic(0xc90280, 0xc00019a6d0)
        /home/marc/sdk/go1.14/src/runtime/panic.go:967 +0x15d
reflect.Value.assignTo(0xcbec20, 0xc00019a670, 0x194, 0xdc13e2, 0xb, 0xce66c0, 0xc00019a680, 0xc000073920, 0xd1b0c0, 0xc000073920)
        /home/marc/sdk/go1.14/src/reflect/value.go:2403 +0x426
reflect.Value.Set(0xce66c0, 0xc00019a680, 0x194, 0xcbec20, 0xc00019a670, 0x194)
        /home/marc/sdk/go1.14/src/reflect/value.go:1532 +0xbd
github.com/containous/yaegi/interp.arrayLit.func1(0xc0001315e0, 0xc000043e00)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:1586 +0x100
github.com/containous/yaegi/interp.runCfg(0xc000138b00, 0xc0001315e0)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:119 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc000376000, 0xc000138300, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:97 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000376000, 0xc0001001e0, 0x5d, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:385 +0x3e8
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:94 +0x548

```

--- a/_test/interface27.go
+++ b/_test/interface27.go
@@ -1,0 +1,12 @@
+package main
+
+import "fmt"
+
+var errs = map[int]error{0: nil}
+
+func main() {
+	fmt.Println(errs)
+}
+
+// Output:
+// map[0:<nil>]

--- a/_test/interface28.go
+++ b/_test/interface28.go
@@ -1,0 +1,12 @@
+package main
+
+import "fmt"
+
+var errs = []error{nil}
+
+func main() {
+	fmt.Println(errs)
+}
+
+// Output:
+// [<nil>]

--- a/interp/run.go
+++ b/interp/run.go
@@ -2393,14 +2393,16 @@ func recv2(n *node) {
 }
 
 func convertLiteralValue(n *node, t reflect.Type) {
-	// Skip non-constant values, undefined target type or interface target type.
-	if !(n.kind == basicLit || n.rval.IsValid()) || t == nil || t.Kind() == reflect.Interface {
-		return
-	}
-	if n.rval.IsValid() {
+	switch {
+	case n.typ.cat == nilT:
+		// Create a zero value of target type.
+		n.rval = reflect.New(t).Elem()
+	case !(n.kind == basicLit || n.rval.IsValid()) || t == nil || t.Kind() == reflect.Interface:
+		// Skip non-constant values, undefined target type or interface target type.
+	case n.rval.IsValid():
 		// Convert constant value to target type.
 		n.rval = n.rval.Convert(t)
-	} else {
+	default:
 		// Create a zero value of target type.
 		n.rval = reflect.New(t).Elem()
 	}

